37 off instrument geometry

michaeljturner opened the pull request:
This feature gives `LoadEmptyInstrument` the capability to load new-style Nexus Geometry files, which are a complete alternative to IDF based instrument creation. This feature is meant as a demonstrator, but the `NexusGeometry` library that comes with this should easily support future adoption of this format.

Despite the number of changes, given that that all existing behaviour of `LoadEmptyInstrument` is still supported, this feature should be low risk.

**Tester**
* Code Review (I appreciate that is going to be tricky with this volume of changes)
* Ensure that the demonstrator nexus files can be loaded via `LoadEmptyInstrument` 
* Highlight any regressions in existing unit or system tests (there should be absolutely none)

OwenArnold commented on Framework/DataHandling/src/LoadNexusGeometry.cpp:
```diff
@@ -0,0 +1,75 @@
+/** Initialize the algorithm's properties.
+ */
+void LoadNexusGeometry::init() {
+  declareProperty("Filename" "", "Name of OFF Nexus Geometry File");
```
Should use `API::FileProperty`

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/InstrumentAbstractBuilder.h:
```diff
@@ -0,0 +1,46 @@
+        this->iImp().addDetector(detName, detId, position);
+    }
+    ///Sort detectors
+    void sortDetectors(){
```
This method would not fit with `Instrument2.0`

OwenArnold commented on Framework/DataHandling/src/LoadNexusGeometry.cpp:
```diff
@@ -0,0 +1,75 @@
+    std::string instName = getProperty("InstrumentName");
+    auto workspace = WorkspaceFactory::Instance().create("Workspace2D", 1, 2, 1);
+
+    NexusGeometry::iAbstractBuilder_sptr iAbsBuilder_sptr = std::shared_ptr<NexusGeometry::iAbstractBuilder>(new NexusGeometry::iAbstractBuilder(instName));
```
use `boost::make_shared`. Also confusing that `iAbstractBuilder` can be newed, given that it's supposed to be abstract. Maybe a rename would fix this.

OwenArnold commented on Framework/DataHandling/src/LoadNexusGeometry.cpp:
```diff
@@ -0,0 +1,75 @@
+void LoadNexusGeometry::exec() {
+    std::string fileName = getProperty("Filename");
+    std::string instName = getProperty("InstrumentName");
+    auto workspace = WorkspaceFactory::Instance().create("Workspace2D", 1, 2, 1);
```
I think we would probably want to have the Workspace as an input property? Much like `LoadInstrument`.

OwenArnold commented on Framework/DataHandling/test/LoadNexusGeometryTest.h:
```diff
@@ -0,0 +1,38 @@
+    TS_ASSERT( alg.isInitialized() )
+    TS_ASSERT_THROWS_NOTHING( alg.setPropertyValue("OutputWorkspace", "_unused_for_child") );
+    TS_ASSERT_THROWS_NOTHING( alg.execute(); );
+    TS_ASSERT( alg.isExecuted() );
```
We will need to test aspects of the workspace geometry (via getInstrument etc) here.

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/InstrumentAbstractBuilder.h:
```diff
@@ -0,0 +1,46 @@
+    }
+private:
+    ///Factor out the static_cast
+    iAbstraction& iImp(){return static_cast<iAbstraction&>(*this);}
```
Would probably be better as a member variable. The name isn't great, will need to change that.

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/InstrumentGeometryAbstraction.h:
```diff
@@ -0,0 +1,40 @@
+namespace Mantid{
+namespace NexusGeometry{
+
+class DLLExport InstrumentGeometryAbstraction : public NexusGeometry::InstrumentAbstractBuilder<InstrumentGeometryAbstraction>{
```
Name of this type needs to change. This is no longer an "Abstraction". `InstrumentBuilder` would be fine. Maybe we'll call the `Instrument2.0` one `BeamlineBuilder`

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/ShapeAbstractCreator.h:
```diff
@@ -0,0 +1,30 @@
+    } 
+private:
+    ///Factor out the static_cast
+    shapeAbstractionClass& shapeImp(){return static_cast<shapeAbstractionClass&>(*this);}
```
Same as above. Make method.

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/ShapeGeometryAbstraction.h:
```diff
@@ -0,0 +1,34 @@
+typedef boost::shared_ptr<Geometry::Object> objectHolder;
+typedef boost::shared_ptr<Geometry::Surface> surfaceHolder;
+
+class DLLExport ShapeGeometryAbstraction : public ShapeAbstractCreator<ShapeGeometryAbstraction, objectHolder>
```
This class is not an abstraction.

OwenArnold commented on Framework/NexusGeometry/src/InstrumentGeometryAbstraction.cpp:
```diff
@@ -0,0 +1,70 @@
+
+    //Default view
+    std::string defaultViewAxis = "z";
+    Geometry::PointingAlong pointingUp(Geometry::Y), alongBeam(Geometry::Z), thetaSign(Geometry::X);
```
I'm not sure about hard-coding reference frame. It depends on what the Nexus standard does about this.

matthew-d-jones replied:
This is the NeXus reference frame, unlike IDFs the NeXus standard doesn't allow you to define another reference frame.

OwenArnold commented on Framework/NexusGeometry/test/InstrumentGeometryAbstractionTest.h:
```diff
@@ -0,0 +1,79 @@
+        iVisitor.walkInstrument();
+        auto iCompInfo = iVisitor.componentInfo();
+        TS_ASSERT(iCompInfo->hasSample());
+        TS_ASSERT(iCompInfo->samplePosition() == this->testPos1);
```
Change these to `TS_ASSERT_EQUALS`

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/InstrumentGeometryAbstraction.h:
```diff
@@ -0,0 +1,40 @@
+        ///Add source 
+        void addSource(std::string &sourceName, Eigen::Vector3d &position);
+        ///Returns underlying instrument
+        Geometry::Instrument_sptr _unAbstractInstrument(){return this->instrument_sptr;}
```
We should think carefully if we really want to unAbstract after all the effort to Abstract.

OwenArnold commented on Framework/NexusGeometry/src/NexusGeometryParser.cpp:
```diff
@@ -0,0 +1,453 @@
+}
+
+///Choose what shape type to parse
+void NexusGeometryParser::parseNexusShape(Group &detectorGroup)
```
Not sure why this isn't returning a Shape. Seems like it ought to.

OwenArnold commented on Framework/NexusGeometry/src/NexusGeometryParser.cpp:
```diff
@@ -0,0 +1,453 @@
+    for(int i = 0; i < 3; ++i){
+        vSorted.col(cPoints[i]) = vertices.col(i);
+    }
+    this->shape = this->sAbsCreator.createCylinder(vSorted);
```
Would much prefer that this returned rather than set a member variable.

OwenArnold reviewed (commented):
First round of review. Clearly Mike is not going to make these changes. This is not production quality yet. Remaining question of prototype->review->dispose as opposed to fix and use.

robertapplin committed:
refs #21465 Export Sample::getShape()

robertapplin committed:
refs #21465 Export Sample::getShape()

robertapplin committed:
refs #21465 Export IObject::getShapeXML

robertapplin committed:
refs #21465 Remove unwanted comment

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/InstrumentBuilder.h:
```diff
@@ -50,10 +53,8 @@ class MANTID_NEXUSGEOMETRY_DLL InstrumentBuilder {
                                      const Eigen::Vector3d &position);
 
   void addObjComponentAssembly(
```
This is really for tubes, should the name reflect this?

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/InstrumentBuilder.h:
```diff
@@ -50,10 +53,8 @@ class MANTID_NEXUSGEOMETRY_DLL InstrumentBuilder {
-      boost::shared_ptr<const Mantid::Geometry::IObject> shape,
-      boost::shared_ptr<const Mantid::Geometry::IObject> detShape,
-      std::vector<Eigen::Vector3d> detPositions, std::vector<int> detIDs);
+      const std::string &compName, const detail::Tube &tube,
```
👍

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/Tube.h:
```diff
@@ -0,0 +1,52 @@
+}
+
+namespace NexusGeometry {
+namespace detail {
```
I really think this type is going to be very confusing without a description. `Tube` is more a builder than an immutable Tube type.

OwenArnold commented on Framework/NexusGeometry/src/Tube.cpp:
```diff
@@ -0,0 +1,104 @@
+
+namespace detail {
+
+Tube::Tube(const Mantid::Geometry::IObject &firstDetectorShape,
```
Calling it firstDetectorShape has more to do with the client side of the usage I think. I think we just want to say that that Tube is operating with some fixed shape for a cylindrical pixel. Should we just call this `pixelShape`.

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/Tube.h:
```diff
@@ -0,0 +1,52 @@
+  double m_height;
+  Eigen::Vector3d m_halfHeightVec; // stored for faster calculations
+  Eigen::Vector3d m_baseVec;       // stored for faster calculations
+  double m_radius;
```
`m_radius` is for the detector pixel (although also same for tube).

OwenArnold commented on Framework/NexusGeometry/inc/MantidNexusGeometry/Tube.h:
```diff
@@ -0,0 +1,52 @@
+private:
+  Eigen::Vector3d m_axis;
+  double m_baseHeight;
+  double m_height;
```
I think this is the height of the pixel not the Tube. Needs to be named & commented as such.

LamarMoore replied:
This is the tube height not the pixel height. `m_baseHeight` is the pixel height.

OwenArnold commented on Framework/NexusGeometry/src/Tube.cpp:
```diff
@@ -0,0 +1,104 @@
+  m_p1 = m_axis + firstDetectorPosition;
+  m_p2 = firstDetectorPosition;
+
+  m_height = m_baseHeight;
```
Don't understand why we have both members?

LamarMoore replied:
Not making assumptions about whether or not there are spaces between tubes. The colinear check does not check whether or not cylinders are touching.

OwenArnold commented on Framework/NexusGeometry/src/Tube.cpp:
```diff
@@ -0,0 +1,104 @@
+  return NexusShapeFactory::createCylinder(points);
+}
+
+const double Tube::height() const { return m_height; }
```
Same thing about this being the real height. I might be mistaken about this, but shouldn't this be `n*m_height` if you really want to return the tube height.

LamarMoore replied:
This is the tube height. `n*m_height` is an assumption.

OwenArnold commented on Framework/NexusGeometry/src/Tube.cpp:
```diff
@@ -0,0 +1,104 @@
+const double Tube::radius() const { return m_radius; }
+
+bool Tube::addDetectorIfCoLinear(const Eigen::Vector3d &pos, int detID) {
+  if (checkCoLinear(pos)) {
```
Very minor, but might read better if we had...
```cpp
cont auto isCoLinear = checkCoLinear(pos);
if(isCoLinear) {
...
return isCoLinear;
```

OwenArnold commented on Framework/NexusGeometry/src/TubeHelpers.cpp:
```diff
@@ -0,0 +1,41 @@
+      auto &tube = (*t);
+      if (tube.addDetectorIfCoLinear(positions.col(i), detIDs[i])) {
+        newEntry = false;
+        break;
```
Comment: Adding detector to existing tube.

OwenArnold commented on Framework/NexusGeometry/src/TubeHelpers.cpp:
```diff
@@ -0,0 +1,41 @@
+  tubes.emplace_back(shape, positions.col(0), detIDs[0]);
+  bool newEntry;
+
+  for (size_t i = 1; i < detIDs.size(); ++i) {
```
Comment: Loop through all detectors and assign to tube(s)

OwenArnold commented on Framework/NexusGeometry/src/TubeHelpers.cpp:
```diff
@@ -0,0 +1,41 @@
+
+  for (size_t i = 1; i < detIDs.size(); ++i) {
+    newEntry = true;
+    for (auto t = tubes.rbegin(); t != tubes.rend(); ++t) {
```
Why reverse iterator. Is this because you expect the last tube to be the most likely candidate of tube to hold the next detector?

LamarMoore replied:
Yes. If the tubes are sorted then this will give the best performance. If they are random it doesn't matter if you iterate forwards or backwards.

OwenArnold commented on Framework/NexusGeometry/src/TubeHelpers.cpp:
```diff
@@ -0,0 +1,41 @@
+namespace NexusGeometry {
+namespace TubeHelpers {
+
+std::vector<detail::Tube> findTubes(const Mantid::Geometry::IObject &shape,
```
Very minor point, is this more a `find` or a `sort`? Leave as is most likely

OwenArnold commented on Framework/NexusGeometry/src/TubeHelpers.cpp:
```diff
@@ -0,0 +1,41 @@
+  }
+
+  // Remove "tubes" with only 1 element
+  tubes.erase(
```
👍 right way to do it, and discussed last week. Just wondering if needed now. Will leave up to you.

LamarMoore replied:
Needed so that a check doesn't need to be shifted somewhere else. Might as well do it here as it doesn't have a massive effect on performance.

OwenArnold commented on Framework/TestHelpers/src/NexusGeometryTestHelpers.cpp:
```diff
@@ -0,0 +1,51 @@
+namespace NexusGeometryTestHelpers {
+boost::shared_ptr<const Mantid::Geometry::IObject> createShape() {
+  Eigen::Matrix<double, 3, 3> pointsDef;
+  pointsDef.col(0) = Eigen::Vector3d(-0.00101, 0.0, 0.0);
```
Could you add comments about what each of these three vectors relates too to help out future readers.

OwenArnold commented on Framework/NexusGeometry/test/TubeTest.h:
```diff
@@ -0,0 +1,65 @@
+
+    TS_ASSERT(!tube.addDetectorIfCoLinear(Eigen::Vector3d(0, 2, 0), 11));
+    TS_ASSERT_EQUALS(tube.size(), 1);
+  }
```
👍

OwenArnold commented on Framework/NexusGeometry/test/TubeHelpersTest.h:
```diff
@@ -0,0 +1,55 @@
+    auto shape = createShape();
+    auto detIds = getFakeDetIDs();
+
+    // replace with invalid coord
```
what do you mean by invalid?

OwenArnold commented on Framework/NexusGeometry/test/TubeHelpersTest.h:
```diff
@@ -0,0 +1,55 @@
+    TS_ASSERT_EQUALS(tubes[0].size(), 2);
+  }
+
+  void test_findTubesValid() {
```
I think this method name does not match the test.

OwenArnold commented on Framework/NexusGeometry/test/TubeHelpersTest.h:
```diff
@@ -0,0 +1,55 @@
+
+class TubeHelpersTest : public CxxTest::TestSuite {
+public:
+  void test_findTubesInvalid() {
```
I think this method name does not match the test.

OwenArnold commented on Framework/NexusGeometry/test/TubeHelpersTest.h:
```diff
@@ -0,0 +1,55 @@
+    auto shape = createShape();
+    auto detIds = getFakeDetIDs();
+
+    auto tubes = TubeHelpers::findTubes(*shape, pixels, detIds);
```
Comment: Inputs represent two parallel arrangements of detector cylinders of two stacked cylinder pixel each.

OwenArnold commented on Framework/NexusGeometry/test/TubeHelpersTest.h:
```diff
@@ -0,0 +1,55 @@
+    TS_ASSERT_EQUALS(tubes.size(), 2);
+    TS_ASSERT(tubes[0].size() == tubes[1].size());
+    TS_ASSERT_EQUALS(tubes[0].size(), 2);
+  }
```
👍

OwenArnold commented on Framework/NexusGeometry/src/NexusGeometryParser.cpp:
```diff
@@ -737,26 +632,25 @@ extractInstrument(const H5File &file, const Group &root) {
+    std::vector<detail::Tube> tubes;
     if (searchTubes) {
-      tubes = findTubes(*shape, pixelOffsets, detectorIds);
+      tubes = TubeHelpers::findTubes(*detShape, pixelOffsets, detectorIds);
```
I don't know if you want to offload to the `builder` to create the whole set of tubes. i.e: `builder.addTubes(bankName, tubes);`

OwenArnold reviewed (dismissed):
Thanks @LamarMoore for the changes. Some relatively minor cleanup required. No major problems spotted.

martyngigg committed:
Limit plot context menu to MatrixWorkspace items

martyngigg committed:
Catch exceptions related to invalid workspace types

BhuvanBezawada committed:
Made copy/move constructors private

re #23145

Commit:
clang-format PR23226 33939ee86

Commit:
Re #23248 fixing flake 8 warning

robertapplin committed:
Remove unnecessary changes in IObject.cpp Refs #21465

peterfpeterson committed:
Change getStoredPropertyValue to getPreviousValue

FitDialog::getStoredPropertyValue was a duplicate version of
AlgorithmDialog::getPreviousValue. This just changes from one to the
other.

peterfpeterson committed:
Move singleton and config into module

Commit:
Split into header and source file Re #23266

nvaytet committed:
Refs #0 : Started modifying systemstests to run with multi-core

peterfpeterson committed:
Put default values into QSettings

peterfpeterson committed:
Set geometry according to QSettings

peterfpeterson committed:
Add method to see if key exists

peterfpeterson committed:
Add method to remove a key

peterfpeterson committed:
Refactor state load/save end fix exit behavior

AndreiSavici committed:
MDNormDirectSC can now handle merged MDEventWorkspaces

AndreiSavici committed:
Clang formatting

robertapplin committed:
Apply clang format patch Refs #21465

martyngigg committed:
Fix warnings coming from the ADS

martyngigg committed:
Minor tidy up of error message

martyngigg committed:
Merge pull request #23270 from rosswhitfield/rpm_postun_$1

Change the check for rpm uninstall/upgrade to use $1

Pasarus committed:
Re #22515 Change location of the histogram comparator

Commit:
Re #23248 cleaning up code to build initial file list

Commit:
Re #23248 moving write_workspace_history function to PyExec

Commit:
Re #23248 removing legacy code

Pasarus committed:
Re #22917 Moved some modified calls to c++ core opposed to export

martyngigg committed:
Fix deprecation warning

martyngigg committed:
Set unique class name for unit test QApplication

martyngigg committed:
Make some python tests quieter

martyngigg committed:
Fix workspace check in Python 3

Pasarus committed:
Re #23248 Removing unused import

peterfpeterson committed:
Add unit test for alternate key

peterfpeterson committed:
Consolidate configuration into less places

BhuvanBezawada committed:
Updated test file for iterator

re #23145

rosswhitfield committed:
Put makeTableWorkspace in anonymous namespace

This should make sure the correct makeTableWorkspace is called

rosswhitfield committed:
Put makeTableWorkspace in anonymous namespace

This should make sure the correct makeTableWorkspace is called

peterfpeterson committed:
Insure that the main window is put on screen

peterfpeterson committed:
Merge pull request #23267 from mantidproject/23266_add_GIL_release

Added GIL release to findRuns python calls

Commit:
Re #23248 removing old commented lines and extraneous debug logs

Commit:
Re #23248 improving comments explaining tokenized tuples

martyngigg committed:
Replace map with for loop

map in Python 2 performed the loop iteration immediately
and returned a list whereas Python 3 returns an iterator
for lazy evaluation. If you don't care about the result
the iteration never happens!

AndreiSavici committed:
doxygen warnings

martyngigg committed:
Decay to Jet cmap if Viridis is not available

Older versions of matplotlib don't ship with Viridis...

jmborr committed:
Refs #23175 Implement reviewer's comments

martyngigg committed:
Use a fixed number of omp threads for unit tests on builders

The default in conjunction with ctest -j produces a large number
of threads that compete with each other and give worse performance.

martyngigg committed:
Use default number of threads for doc tests

martyngigg committed:
Clarify documentation regarding Number property

BhuvanBezawada committed:
Refactored iterator class and item class

re #23278

robertapplin committed:
Remove unused 'using CSGObject' statement Refs #21465

Commit:
Re #22406 Child algorithms rethrow, so no need to check isExecuted

Pasarus committed:
Re #23280 Initial Commit

Pasarus committed:
Re #23280 Clang format

peterfpeterson committed:
AlgorithmInputHistory is a child of Configurable

martyngigg committed:
Merge pull request #23243 from gemmaguest/fix_layout_bug_on_refl_gui

martyngigg committed:
Merge pull request #23176 from mantidproject/23175_AppendSpectra

Enable appending multiple times for event workspaces

martyngigg committed:
Merge pull request #23271 from mantidproject/FitDialog_duplicate_code

Change getStoredPropertyValue to getPreviousValue

martyngigg committed:
Merge pull request #23281 from BhuvanBezawada/23278_refactor_histogram_iterator_test

Refactored `HistogramIterator` and  `HistogramItem`

nvaytet committed:
Refs #23285 : Now using a separate xml reporter for each instance.

martyngigg committed:
Add missing createSuite/destroySuite statements

martyngigg committed:
Merge pull request #23273 from mantidproject/workbench_use_config

Workbench use QSettings configuration

quantumsteve committed:
Merge pull request #23250 from martyngigg/workbench_dragndrop_plotting

Workbench dragndrop plotting

AndreiSavici committed:
Merge pull request #23282 from mantidproject/23280_AddisGroup()ToWorkspaces

Add isGroup() method to Workspace.h and WorkspaceGroup.h

AndreiSavici committed:
Merge pull request #23272 from rosswhitfield/fix_mantid.dataobjects.TableWorkspace_clang

Fix mantid.dataobjects.TableWorkspace() for Linux clang

peterfpeterson committed:
Expose AlgorithmInputHistory to python

peterfpeterson committed:
Read and write algorithm dialog values

Commit:
Re #23248 rewrote to avoid un-necessary reconstruction of data

peterfpeterson committed:
Merge pull request #23096 from mantidproject/21465_sample_shape_export

Export Sample::getShape()

peterfpeterson committed:
Merge pull request #22940 from BhuvanBezawada/22923_expose_component_info

Exposed most of the methods from ComponentInfo

peterfpeterson committed:
Merge pull request #22961 from mantidproject/22917_setCellUpdateMantidPlot

ITableWorkspace now updates GUI when data changed via Python

martyngigg committed:
Fix memory leaks reported from KernelTest

martyngigg committed:
Fix memory leaks reported from GeometryTest

martyngigg committed:
Fix out-of-bounds access in Geometry/Crystal

martyngigg committed:
Fix input array dimensions for NeXus file write

martyngigg committed:
Use std::copy over vector::assign to avoid overflow

martyngigg committed:
Fix test input values and check for bad input.

martyngigg committed:
Release Poco::XML document node at the end of processing

martyngigg committed:
Use unique_ptr over raw new/delete

Fixes a memory leak on the exception path.

martyngigg committed:
Use unique_ptr to manage array memory

Avoid memory leaks on various error paths.

martyngigg committed:
Add boilerplate code to all ICat tests

Also replaces CatalogDownloadDataFilesTest::skipTests to use
the common skipTests implementation.

Commit:
Re #23248 cleaning print statement and tuple build

MikeHart85 committed:
Merge branch 'master' into 22406_loadlivedata_bin_boundaries

gemmaguest committed:
Merge branch 'master' into 23045_offspec_parameters_file

OwenArnold committed:
Merge branch 'master' of https://github.com/mantidproject/mantid into 37_off_instrument_geometry

martyngigg committed:
Merge pull request #23288 from mantidproject/AlgorithmHistory_in_python

Algorithm history in python

OwenArnold committed:
Clang format -> 5.0

OwenArnold committed:
Reduce scope of variable

OwenArnold committed:
Fixing warnings and errors

OwenArnold committed:
Clang formatting

peterfpeterson committed:
Merge pull request #23277 from martyngigg/limit-omp-threads-builders

Use a fixed number of omp threads for unit tests on builders

OwenArnold committed:
Make more portable

OwenArnold committed:
Add Config file path features

OwenArnold committed:
Formatting

gemmaguest committed:
Minor change to make valid-to date more sensible

Updated the valid-to date to be the same as the valid-from date for the experimental IDF to make it clearer that this IDF is never used at the moment

Re #16218

nvaytet committed:
Refs #23285 : Distributing list of tests module by module

This is done to avoid cleanup failures. Some data files were being deleted by another process
while the current process running a test in the same module using the same file had not yet finished.

peterfpeterson committed:
Correct main window location

Refs #23284

peterfpeterson committed:
Merge remote-tracking branch 'martyn/kerneltest-address-sanitizer'

peterfpeterson committed:
Remove unused import

peterfpeterson committed:
Merge pull request #23287 from martyngigg/missing-unittest-boilerplate

Add missing unit test boilerplate

Commit:
Add tests cases for LoadGudrunOutput

Refs #23298

nvaytet committed:
Refs #23285 : Printing results in one go as a large string

AndreiSavici committed:
no need to verify

AndreiSavici committed:
clang formatting

OwenArnold committed:
Merge loading into LoadEmptyInstrument

Remove LoadEmptyNexusGeometry algorithm.

martyngigg committed:
Merge pull request #23199 from gemmaguest/16218_offspec_and_crisp_IDFs

Update OFFSPEC and CRISP IDFs

Pasarus committed:
Re #23248 Update release notes and fix Flake8

Pasarus committed:
Re #23248 Remove trailing whitespace

SimonHeybrock committed:
Merge pull request #22852 from mantidproject/22515_SortXAxisinc++

Add C++ version of SortXAxis #22515

LamarMoore committed:
Merge pull request #22884 from BhuvanBezawada/export_spectrum_definition

Exported SpectrumInfo and SpectrumDefinition

ewancook committed:
refs #23023 update as per review, naming consistency, replace assert

ewancook committed:
refs #23023 add view tests

Commit:
Improve tests and add data files

Refs #23298

Commit:
Implement LoadGudrunOutput alg

Refs #23298

Commit:
Update documentation for release notes and help

Refs #23298

OwenArnold committed:
Fix bad merge

AnthonyLim23 committed:
refs #23232 rm globbal syntax error

AnthonyLim23 committed:
refs #23232 rm globbal syntax error 2

AndreiSavici committed:
Merge pull request #23047 from mantidproject/23045_offspec_parameters_file

Add polarization correction data to parameters file

AndreiSavici committed:
Merge pull request #23222 from mantidproject/23115_elemental_analysis_menu_bar

Elemental Analysis Menu Bar

nvaytet committed:
Refs #23285 : Trying to collect list of failed tests in array

AnthonyLim23 committed:
refs #23232 added global onto previous lines

Pasarus committed:
Re #22896 Implement SortXAxis in Stitch1D

OwenArnold committed:
Better file detection in LoadInstrument

Commit:
make python 3 compatiable

Refs #23298

Commit:
Re #22406 Reverted monitor workspace copy fix

VickieLynch committed:
Refs #23301 sort before numbering peaks

peterfpeterson committed:
Add support for fat events

This was done by adding two more properties which allow specifying the
wall-clock tolerance. If those properties aren't set then the old
behavior is retained.

nvaytet committed:
Refs #23285 : Summary from Manager.dict ok

SimonHeybrock committed:
Merge pull request #23300 from mantidproject/workbench_placement

Correct workbench window location

SimonHeybrock committed:
Merge pull request #23252 from mantidproject/23248_SanitiseThe#CharecterProjectRevocery

Project Recovery: Sanitise the # charecter in workspace names

SimonHeybrock committed:
Merge pull request #23245 from mantidproject/23232_MuonPYRewrite

Restructure of  muon analysis 2 py file for saving

SimonHeybrock committed:
Merge pull request #23205 from mantidproject/23149_polaris_van_multi_load

ISIS Powder POLARIS load vanadium lifted out of loop

nvaytet committed:
Refs #23285 : Parallel systemtests seem to work now

Use the '-j 8' option to run on 8 cpus.

There are now two different modes of console output:
- the full mode where all output from the tests are printed (this is the old behaviour)
- the quiet mode (activated by the '-q' or '--quiet' option) that just prints the progress

BhuvanBezawada committed:
Fixed line spacing error

re #23189

nvaytet committed:
Merge branch 'master' into 20443_CreateWorkspace_slow

OwenArnold committed:
Get around throwing constructor

The ..descriptor types throw if files are invalid.

nvaytet committed:
Refs #23285 : Attempt to run on multiple cores on built server

BhuvanBezawada committed:
Merge branch 'master' into 23189_release_notes_for_all_info_exports

re #23189

nvaytet committed:
Refs #20443 : Reverted to OMP schedule(static)

BhuvanBezawada committed:
Removed getIndex() from public interface

re #23145

nvaytet committed:
Refs #20443 : Comment length

Pasarus committed:
Re #22896 Utilise Child algorithms

Commit:
ReflectometrySumInQTest: check the result, not just if it throws.

Re #22723

AndreiSavici committed:
Allow MDNormSCD to work on merged workspaces

AndreiSavici committed:
Clang format

LamarMoore committed:
Fix det relative positions and failing unit tests

peterfpeterson committed:
Merge pull request #23274 from mantidproject/MDNorm_merged

MDNormDirectSC can now handle merged MDEventWorkspaces

peterfpeterson committed:
Merge pull request #22666 from mantidproject/20443_CreateWorkspace_slow

Refs #20443 : Threading of Strings::join()

AndreiSavici committed:
Merge pull request #23312 from mantidproject/AlignAndFocus_weighted_events

Add support for fat events to AlignAndFocusPowder

BhuvanBezawada committed:
Merge branch 'master' into 23145_iterator_for_detector_info

OwenArnold committed:
Merge branch 'master'

of https://github.com/mantidproject/mantid into 37_off_instrument_geometry

Commit:
ReflectometrySumInQ: Use deg2rad and rad2deg from Geometry. Re #22723

Commit:
Merge branch 'refs/heads/master' into
22723_ReflectometrySumInQ_unit_test

OwenArnold committed:
Fix narrowing warnings

martyngigg committed:
Use std::vector over std::unique_ptr<std::vector>

OwenArnold committed:
Fix doctest

Commit:
Fix build. Change Geometry::deg2rad etc. const -> constexpr Re #22723

martyngigg committed:
Improve readability of loop stop condition

martyngigg committed:
Re-enable test that was disabled by accident

Pasarus committed:
Re #22896 Remove unused imports

martyngigg committed:
Obey skipTests method in dynamic test suites

rosswhitfield committed:
Merge pull request #23326 from martyngigg/icat-skiptests-dynamic-suites

Obey skipTests method in dynamic CxxTest test suites

OwenArnold committed:
Release notes and other docs

martyngigg committed:
Merge pull request #23311 from mantidproject/22896_ReplaceSortXAxisInStitch1D

Implementing SortXAxis in Stitch1D

martyngigg committed:
Merge pull request #22724 from mantidproject/22723_ReflectometrySumInQ_unit_test

Improve ReflectometrySumInQ unit tests

peterfpeterson committed:
Add parameter name to exception

peterfpeterson committed:
Merge remote-tracking branch 'origin/master' into ConfigService_doc_prop

peterfpeterson committed:
Add MantidOptions.ReusePlotInstances to ignore

VickieLynch committed:
Refs #23301 sort by bank name

VickieLynch committed:
Refs #23301 fix doc test

BhuvanBezawada committed:
Updated release notes with hyperlinks

re #23189

BhuvanBezawada committed:
Moved DetectorInfoPythonIterator into PythonInterface

re #23145

BhuvanBezawada committed:
Merge branch 'master' into 23145_iterator_for_detector_info

re #23145

BhuvanBezawada committed:
Merge branch 'master' into 23145_iterator_for_detector_info

re #23145

BhuvanBezawada committed:
Merge branch '23145_iterator_for_detector_info' of https://github.com/BhuvanBezawada/mantid into 23145_iterator_for_detector_info

re #23145

Commit:
clang-format PR23226 c31fd0ee4

nvaytet committed:
Refs #23285 : Now doing a round of cleanup at the end

nvaytet committed:
Refs #23285 : Fixed syntax error

BhuvanBezawada committed:
Minor changes for hyperlinks to full docs

re #23189

BhuvanBezawada committed:
Removed unnecessary links to concepts docs

re #23189

peterfpeterson committed:
Alphabetize properties within the tables

peterfpeterson committed:
Merge pull request #23291 from martyngigg/datahandling-address-sanitizer

Fix issues in DataHandlingTest reported by address sanitizer

peterfpeterson committed:
Merge pull request #23290 from martyngigg/geometrytest-address-sanitizer

Fix issues with GeometryTest address identified by address sanitizer

peterfpeterson committed:
Merge pull request #23319 from mantidproject/MDNormSCD_merged

Allow MDNormSCD to work on merged workspaces

rosswhitfield committed:
Merge pull request #23331 from mantidproject/ImprovedExceptionMessage

Add parameter name to exception

rosswhitfield committed:
Merge pull request #23194 from mantidproject/ConfigService_doc_prop

ConfigService document more properties

AndreiSavici committed:
Merge pull request #23333 from mantidproject/23301_peak_number

Sort peak numbers by detector banks

Commit:
Merge pull request #23254 from mantidproject/23023_run_data_loading

Elemental Analysis - Run Data Loading

Commit:
Merge pull request #22425 from mantidproject/22406_loadlivedata_bin_boundaries

LoadLiveData should adjust bin boundaries to fit all events when in PreserveEvents mode

peterfpeterson committed:
Throw ParseError instead of bad_lexical_cast

Users reported it difficult to find where errors were in files. This
should greatly improve their ability to figure out what is going on.

peterfpeterson committed:
Add instruction about devtoolset-7

peterfpeterson committed:
Fix code-block formatting

rosswhitfield committed:
Merge pull request #23350 from mantidproject/build_instructions

Add instruction about devtoolset-7

ElliotAOram committed:
Merge branch 'master' into 23298-gudrun-output-loader

martyngigg committed:
Merge pull request #23226 from BhuvanBezawada/23145_iterator_for_detector_info

Implemented an iterator for `DetectorInfo`

OwenArnold committed:
Remove dead lines

OwenArnold committed:
MeshObject is 3D

OwenArnold committed:
Delete comment lines

OwenArnold committed:
Consistent leading whitespace

SimonHeybrock committed:
Merge pull request #23206 from BhuvanBezawada/23189_release_notes_for_all_info_exports

Updated release notes for info exports

SimonHeybrock committed:
Merge pull request #23210 from mantidproject/gui_usability_fixes

Fixed run box selection issue and SampleShape change

OwenArnold committed:
Use specific NX{}_geometry classes

nvaytet committed:
Refs #23285 : Re-arranged lists of tests and edited Jenkins files

Some tests were still failing because different modules are writing in the same directories, using the same files names.
There are at least 3 different locations the tests write to:
1. build/ExternalData/Testing/Data/SystemTest
2. build/Testing/SystemTests/scripts
3. build
In some cases a process tries to write to a file that already exists (same name used by another test).
In other cases a process performs a .cleanup() which removes a file that is still needed by a running test.
The solution was to group the tests in lists where all modules starting with the same 4 letters are given to one
core. This worsens the load balance between cores (with 8 cores, core 1 performs 93 tests while core 8 only has 44).
This is not ideal but allows the suite to complete without failures.
The runtime using 8 cores goes down from 2h to 30min, which is ok as a first step.

OwenArnold committed:
No need to make 2D -> 3D in MeshObject

martyngigg committed:
Fix workbench plotting test for matplotlib >= v2

Axes.get_lines() changed its behaviour when error bars are plotted
in version 2. This change covers both versions.

OwenArnold committed:
Simplify coplanar check

nvaytet committed:
Refs #23285 : Got rid of some duplicate variables

nvaytet committed:
Refs #23285 : SystemTests in quiet mode to reduce log size

OwenArnold committed:
Tolerance on distance calculation

BhuvanBezawada committed:
Removed duplicated exports

re #23358

OwenArnold committed:
More cleanup following review

mantid-roman committed:
Merge remote-tracking branch 'origin/master' into 23009-debug-reflectometry-search-bug

Re #23009

nvaytet committed:
Refs #23285 : Fixed bug in InstallerTests arguments

OwenArnold committed:
More cleanup post review

Commit:
Merge pull request #23355 from martyngigg/mpl-v2-unit-tests

Fix workbench plotting test for matplotlib >= v2

OwenArnold committed:
No need for Transaction

OwenArnold committed:
Various improvements to NexusGeometryParser

OwenArnold committed:
Parsing error types not needed

OwenArnold committed:
More changes from review

peterfpeterson committed:
Update documentation for external data

peterfpeterson committed:
Merge pull request #23360 from BhuvanBezawada/23358_fix_multiple_exports_error

Removed duplicated exports

nvaytet committed:
Refs #23285 : Fixed j= error in InstallerTests.

OwenArnold committed:
Fix complex transform test

martyngigg committed:
Remove reliance on peak ordering in WISH system test

The test required that a found peak was in a specific
position in the table but this is not guarnateed any longer
after bd4b4e3c88ec72e5565d4472a012395ef0d0a08f. The test failure
was not spotted on the original pull request as it was skipped
due to memory limitations on the node.

OwenArnold committed:
Compare real instrument idf, vs nexus

Also constains fix for monitor transformations

OwenArnold committed:
Check monitor positions in testing

OwenArnold committed:
Validate that instruments are different a prori

OwenArnold committed:
Compare base instruments

OwenArnold committed:
InstrumentBuilder resets internal instrument

OwenArnold committed:
Use std::make_unique

nvaytet committed:
Refs #23285 : Print warning only when global test list is empty

nvaytet committed:
Refs #23285 : Updated developer documentation

nvaytet committed:
Refs #23285 : Fixed code blocks in dev docs

peterfpeterson committed:
Merge pull request #23370 from mantidproject/wish_sc_system_test_ordering

Remove reliance on peak ordering in WISH SC system test

peterfpeterson committed:
Merge pull request #23249 from mantidproject/20525_LiveListeners_LoadInstrument_repeatedly

Avoid multiple loading of instrument data in ISISHistoDataListener

martyngigg committed:
Merge pull request #23348 from mantidproject/PDLoad_parse_error

Throw ParseError instead of bad_lexical_cast

AndreiSavici committed:
Merge pull request #23365 from mantidproject/external_data_doc

Update documentation for external data

AndreiSavici committed:
Merge pull request #23010 from mantidproject/23009-debug-reflectometry-search-bug

Reflectometry search debug assertion failure

nvaytet committed:
Refs #23285 : Merge branch 'master' into multicore_systemtests

peterfpeterson committed:
Merge pull request #23305 from mantidproject/23298-gudrun-output-loader

GudrunOutputLoader

nvaytet committed:
Refs #23285 : Fixed python3 error

SimonHeybrock committed:
Merge pull request #23286 from mantidproject/multicore_systemtests

Multicore systemtests

OwenArnold committed:
Further review fixes

OwenArnold committed:
Merge branch 'master'

of https://github.com/mantidproject/mantid into 37_off_instrument_geometry

OwenArnold committed:
Missing boiler plate static methods

OwenArnold committed:
Satisfy older apple clang. unique_ptr<const T>

implicit conversion from unique_ptr<T> missing

OwenArnold committed:
Add comment about fix
